### PR TITLE
PR: Adding integration test 

### PR DESCRIPTION
**Pull request by chrislaughlin**
Adding a basic e2e test using Ember and qunit, this seems to be the standard for ember development and is supported by a lot of people from what I can see. The gulp task that runs the tests uses phantom as the browser however we could expand this to others. This is a start for the e2e tests and with the built in Ember test helpers there seems to be a good framework for e2e testing at least.

I also did some code clean up, removed some self executing functions that were not needed.

### CODE REVIEW

**Comment by chrislaughlin** [comment]
Looks like there are some conflicts that will have to be handled so please merge down and check the code out.

File: client/src/app/controllers/loginController.js [hunk 1]
```diff
@@ -1,43 +1,41 @@
...
+            $.post('/auth.json', data).then(function (response) {
 
-                    self.set('errorMessage', response.message);
-                    if (response.success) {
-                        alert('Login succeeded!');
-                        self.set('token', response.token);
+                self.set('errorMessage', response.message);
+                if (response.success) {
+                    alert('Login succeeded!');
+                    self.set('token', response.token);
```
Comment (martinmicunda) [inline client/src/app/controllers/loginController.js:14]: @chrislaughlin are we using token or session for authentication. Where is the server code for auth?

**Comment by martinmicunda** [comment]
Chris you can merge the code and fix the conflict :)

File: client/src/app/controllers/loginController.js (hunk 1 above)
Comment (chrislaughlin) [inline client/src/app/controllers/loginController.js:14]: The code is using dummy data in /server/src/routes/index.js

**Comment by chrislaughlin** [comment]
Trying now lol

File: client/src/app/controllers/loginController.js (hunk 1 above)
Comment (martinmicunda) [inline client/src/app/controllers/loginController.js:14]: ah ok...

**Comment by chrislaughlin** [comment]
merged

File: client/src/app/controllers/loginController.js (hunk 1 above)
Comment (chrislaughlin) [inline client/src/app/controllers/loginController.js:14]: @martin can you pull the code and check everything is okay

On Thursday, May 29, 2014, Martin Micunda notifications@github.com wrote:

> In client/src/app/controllers/loginController.js:
> 
> > -                    self.set('errorMessage', response.message);
> > -                    if (response.success) {
> > -                        alert('Login succeeded!');
> > -                        self.set('token', response.token);
> > -                self.set('errorMessage', response.message);
> > -                if (response.success) {
> > -                    alert('Login succeeded!');
> > -                    self.set('token', response.token);
> 
> ah ok...
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/cam-technologies/time-booker/pull/19/files#r13203906
> .

## 

Regards

Chris Laughlin

Comment (martinmicunda) [inline client/src/app/controllers/loginController.js:14]: @chrislaughlin yeah it's working fine.. btw there is no option logout

Comment (chrislaughlin) [inline client/src/app/controllers/loginController.js:14]: Yeah once we get all the tests done we can work on dev stuff

On Thursday, May 29, 2014, Martin Micunda notifications@github.com wrote:

> In client/src/app/controllers/loginController.js:
> 
> > -                    self.set('errorMessage', response.message);
> > -                    if (response.success) {
> > -                        alert('Login succeeded!');
> > -                        self.set('token', response.token);
> > -                self.set('errorMessage', response.message);
> > -                if (response.success) {
> > -                    alert('Login succeeded!');
> > -                    self.set('token', response.token);
> 
> @chrislaughlin https://github.com/chrislaughlin yeah it's working
> fine.. btw there is no option logout
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/cam-technologies/time-booker/pull/19/files#r13205087
> .

## 

Regards

Chris Laughlin

**Pull request closed without merging**